apps/www/components/Pledge/Submit.js

```diff
@@ -172,6 +236,110 @@ class Submit extends Component {
+
+    // Since we don't want to always load all Payment Providers data,
+    // we must reinitialize when switching to a different wallet
+    const isOutdatedStripeWallet =
+      this.isStripeWalletPayment() &&
+      this.props.paymentRequest.status !== PaymentRequestStatus.IDLE &&
+      this.props.paymentRequest.usedWallet !== this.props.selectedPaymentMethod
+
+    const isUnavailableAndStripeWalletIsSelected =
+      this.isStripeWalletPayment() &&
+      this.props.paymentRequest.status === PaymentRequestStatus.UNAVAILABLE
+
+    if (isUninitializedStripeWallet || isOutdatedStripeWallet) {
+      const selectedPaymentMethod = this.props.selectedPaymentMethod
+
+      // Create payment-request
+      this.props.paymentRequest
+        .initialize(selectedPaymentMethod)
+        .then((status) => {
+          if (status !== PaymentRequestStatus.UNAVAILABLE) {
+            this.setState(() => ({
+              loading: false,
+              walletError: null,
+            }))
+          } else {
+            const isGooglePayPayment =
+              this.props.selectedPaymentMethod ===
+              WalletPaymentMethod.GOOGLE_PAY
+
+            if (isGooglePayPayment) {
+              setSelectedPaymentMethod('STRIPE')
+            }
+            this.setState((prevState) => ({
+              ...prevState,
+              loading: false,
+              walletError: !isGooglePayPayment
+                ? t('account/pledges/payment/methods/unavailable')
+                : null,
+              stripeError: isGooglePayPayment
+                ? t('account/pledges/payment/methods/google-pay/unavailable')
+                : null,
+            }))
+          }
+        })
+        .catch((error) => {
+          console.error('Wallet initialization error', error)
+          this.setState(() => ({
+            loading: false,
+            walletError: t(
+              'account/pledges/payment/methods/initialization/error',
+            ),
+          }))
+        })
+
+      // Re-initialize the payment-request for the current wallet
+      this.setState(() => ({
+        loading: t('account/pledges/payment/methods/loading'), // The loading state is tracked though paymentRequest.status
+      }))
+    } else if (
+      isUnavailableAndStripeWalletIsSelected &&
+      !this.state.walletError
+    ) {
+      this.setState(() => ({
+        loading: false,
+        walletError: t('account/pledges/payment/methods/unavailable'),
+      }))
+    }
+    if (
+      this.props.selectedPaymentMethod &&
+      !this.props.selectedPaymentMethod.startsWith('STRIPE-WALLET') &&
+      this.state.walletError
+    ) {
+      this.setState(() => ({
+        walletError: null,
+      }))
+    }
+    if (
+      this.props.selectedPaymentMethod &&
+      !this.props.selectedPaymentMethod.startsWith('STRIPE') &&
+      this.state.stripeError
+    ) {
+      this.setState(() => ({
+        stripeError: null,
+      }))
+    }
+  }
```

Comment:
I would have imagined that we move this whole logic to the `usePaymentRequest` hook

Reply:
In case we do that, we would have to tear most of the Submit-component state apart and move it up to Submit Hooks. If we were to do that, we might as well refactor the entire Submit-component and convert it into a functional-component.

Reply:
are you sure? As far as I can see it's only `loading`, `walletError`, `stripeError` – afaik our `PaymentRequestValues` already have a loading state and could be extended with two error states?

Reply:
Done